README.md

```diff
@@ -0,0 +1,88 @@
+# PoC - Reuse OpenTelemetry .NET API and SDK
+
+## Vision
+
+Use [OpenTelemetry .NET API and SDK](https://github.com/open-telemetry/opentelemetry-dotnet)
+instead Datadog tracer model.
+
+## Use Case
+
+Instrumented HTTP client and HTTP server applciations exporting traces to Jaeger.
+We may also try adding manual instrumentation.
+
+## Constraints
+
+We may add additional requirements for the instrumented applications such us:
+
+- Add references to OpenTelemetry NuGet packages.
+- Concrete versions of OpenTelemetry NuGet packages must be used.
+- Must use newer versions of .NET.
+
+## Plan
+
+### Preparation
+
+1. Prepare the use case secenario and make sure it is working.
+1. Get rid of everything, but HTTP Client and HTTP Server instrumentation.
+   Reason: quicker iterations (especially thanks to quicker build time).
+   **Make sure the use case is still working**.
+
+It would be good to have some automation here to test without doing to much stuff manually.
+
+### Remove Datadog Tracer
+
+1. Remove `Datadog.Trace.Tracer` and a lot of related code.
+   We can just keep the AutoInstrumentation integration boilerplate for HTTP Client and ASP.NET.
+1. Change the `Datadog.Trace.ClrProfiler.Instrumentation.Initialize()` method
+   so that it will be the place where the global OTel trace provider will be configured. 
+
+It would be good to have a checkpoint here where the instrumented application is NOOP-instrumented.
+
+## Experiments
+
+1. Hardcode the global OTel tracer configuration.
+1. Configure the OTel tracer based on env vars. SDK should handle it.
+   [question](https://cloud-native.slack.com/archives/C01N3BC2W7Q/p1620994235161800)
+1. Add manual instrumentation to the applications.
+1. Try auto-instrumentation using ActivitySource
+   e.g. in `Datadog.Trace.ClrProfiler.AutoInstrumentation.Http.HttpClient.HttpMessageHandlerCommon`.
+1. Test with different versions of OTel SDK in runtime.
+1. Try using `AppDomain.CurrentDomain.AssemblyResolve +=` to add a fallback 
+   if the instrumented app does not reference the OTel SDK.
+
+## Helpful docs and samples
+
+- https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry/README.md#activity-source
+- https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/examples/Console/TestHttpClient.cs
+
+## Results
+
+## Testing
+
+### Additional steps for Windows
```

Comment:
This is probably for Linux too but before my first try I cleaned up the repo so I got an error about missing nuget packages, so I had to run ` $ nuget restore Datadog.Trace.minimal.sln` then everything worked smoothly. 👍

Reply:
Here is the error:
```terminal
Project "C:\otel.net.instr\src\Datadog.Trace.ClrProfiler.Native\Datadog.Trace.ClrProfiler.Native.DLL.vcxproj" (2) is building "C:\otel.net.instr\src\Datadog.Trace.ClrProfiler.Native\Datadog.Trace.ClrProfiler.Native.vcxproj" (3) on node 1 (default targets).
C:\otel.net.instr\src\Datadog.Trace.ClrProfiler.Native\Datadog.Trace.ClrProfiler.Native.vcxproj(240,5): error : This project references NuGet package(s) that are missing on this computer. Use NuGet Package Restore to download them.  For more information, see http://go.microsoft.com/fwlink/?LinkID=322105. The missing file is ..\..\packages\nlohmann.json.3.5.0\build\native\nlohmann.json.targets.
Done Building Project "C:\otel.net.instr\src\Datadog.Trace.ClrProfiler.Native\Datadog.Trace.ClrProfiler.Native.vcxproj" (default targets) -- FAILED.
Done Building Project "C:\otel.net.instr\src\Datadog.Trace.ClrProfiler.Native\Datadog.Trace.ClrProfiler.Native.DLL.vcxproj" (Build target(s)) -- FAILED.
Done Building Project "C:\otel.net.instr\Datadog.Trace.proj" (BuildCpp target(s)) -- FAILED.
```